vaadin-combo-box-flow-parent/vaadin-combo-box-flow/src/main/java/com/vaadin/flow/component/combobox/ComboBox.java

```diff
@@ -942,7 +942,7 @@ private void executeRegistration(PropertyChangeEvent event) {
     protected void onAttach(AttachEvent attachEvent) {
         super.onAttach(attachEvent);
         DataProvider<T, ?> dataProvider = getDataProvider();
-        if (dataProvider != null && dataProviderListener == null) {
+        if (dataProvider != null) {
```

Comment:
Unit test please ?

Reply:
Initially, I was thinking that this change is covered by `FilteringIT`. But this filtering test is a separate thing, and it's quite hard to make connections between filtering and DP listeners. So, it's a good idea to add a specific test covering this particular case with DP listeners.

Reply:
The `Grid` has the same code (not leading to a test failures though), so I'll fix it as well.

Reply:
And some other component which use data provider...:)

Reply:
Done